switcheroo/ssh/scripts/publish.py

```diff
@@ -66,26 +74,56 @@ def create_argument_parser() -> ArgumentParser:
     return argument_parser
 
 
-def main():
+def _local_store(sshdir: str, bucket: str | None = None) -> FileKeyPublisher:
+    if bucket is not None:
+        raise InvalidArgumentError(
+            'Invalid argument "--bucket" when storing the keys locally'
+        )
+    return FileKeyPublisher(Path(sshdir))
+
+
+def _s3_store(sshdir: str, bucket: str | None = None) -> S3KeyPublisher:
+    if bucket is None:
+        raise MissingArgumentError("The s3 option requires a bucket name!")
+    return S3KeyPublisher(bucket, root_ssh_dir=Path(sshdir))
+
+
+def _metrics(
+    metricpath: str | None = None, metric: str | None = None
+) -> MetricPublisher:
+    if metric == "file":  # publish to file system
+        return FileMetricPublisher(Path(metricpath))
+    if metric == "aws":  # publish to cloudwatch
+        if metricpath is not None:
+            raise InvalidArgumentError(
+                'Invalid argument "--metricpath" when storing the metrics on AWS'
+            )
+        return AwsMetricPublisher(MetricConstants.NAME_SPACE)
+    raise MissingArgumentError(
+        'Please specify either "file" or "aws" after the -m/--metric option.'
+    )
```

Comment:
Not sure if we need this since it's not adding much - argparse already pretty much says that